PugSharp.Match/Match.cs

```diff
@@ -937,8 +956,61 @@ private MatchPlayer GetMatchPlayer(ulong steamID)
         return AllMatchPlayers.First(x => x.Player.SteamID == steamID);
     }
 
+    public bool PlayerIsReady(ulong steamID)
+    {
+        var matchPlayer = AllMatchPlayers.FirstOrDefault(x => x.Player.SteamID == steamID);
+        return matchPlayer != null && matchPlayer.IsReady;
+    }
+
     #region Match Functions
 
+    private bool TryAddPlayerToCurrentTeam(IPlayer player)
+    {
+        var matchPlayer = MatchInfo.MatchTeam1.Players.Any(x => x.Player.SteamID == player.SteamID) || MatchInfo.MatchTeam2.Players.Any(x => x.Player.SteamID == player.SteamID) ? GetMatchPlayer(player.SteamID) : null;
+        if (matchPlayer == null) {
+            matchPlayer = new MatchPlayer(player);
+        }
+        else {
+            // Remove the player from the match
+            MatchInfo.MatchTeam1.Players.Remove(matchPlayer);
+            MatchInfo.MatchTeam2.Players.Remove(matchPlayer);
+        }
+
+        // Add them back to the match
+        bool? isTeam1 = null;
+        if (MatchInfo.MatchTeam1.CurrentTeamSite == Team.None && MatchInfo.MatchTeam2.CurrentTeamSite == Team.None)
+        {
+            isTeam1 = player.Team == Team.Terrorist;
+        }
+        else if (MatchInfo.MatchTeam1.CurrentTeamSite != Team.None)
+        {
+            isTeam1 = MatchInfo.MatchTeam1.CurrentTeamSite == player.Team;
+        }
+        else if (MatchInfo.MatchTeam2.CurrentTeamSite != Team.None)
+        {
+            isTeam1 = MatchInfo.MatchTeam2.CurrentTeamSite != player.Team;
+        }
+
+        if (isTeam1 == null)
+        {
+            _Logger.LogInformation("Could not add player to match", player.SteamID);
+            return false;
+        }
+        else if (isTeam1.HasValue && isTeam1.Value)
```

Comment:
`isTeam1.HasValue` has not to be checked again. if `isTeam1` is not null HasValue returns always true

Reply:
I'm not sure I follow. The logic is to check if the value is boolean (not null) and true, then add to Team 1, otherwise add to Team 2.